Fulfillment/APIs/In progress orders.md

```diff
@@ -0,0 +1,156 @@
+# In progress Orders API
+
+Fetches a comprehensive list of all orders which are in process of pickup for fulfillment. To get the In progress orders, you will need to call the /solr-query endpoint with the POST method.
+
+## Request
+
+### Endpoint
+
+`https://<host>/api/solr-query`
+
+Example: https://demo-oms.hotwax.io/api/solr-query
+
+### Header
+
+#### Body
+
+```
+{
+  "json": {
+    "params": {
+      "rows": "10",
+      "sort": "orderDate asc",
+      "group": true,
+      "group.field": "picklistBinId",
+      "group.limit": 1000,
+      "group.ngroups": true,
+      "q.op": "AND",
+      "start": 0
+    },
+    "query": "(*:*)",
+    "filter": "docType: OISGIR AND picklistItemStatusId: PICKITEM_PENDING AND -fulfillmentStatus: Rejected AND -shipmentMethodTypeId: STOREPICKUP AND facilityId: Store_1 AND productStoreId: STORE"
+  }
+}
+```
+
+| Parameter                | Description                                                                                                   | Required (Y/N) |
+|--------------------------|---------------------------------------------------------------------------------------------------------------|----------------|
+| `rows`                   | The number of groups.                                                                                        | No.            |
+| `group`                  | Filter to group orders.                                                                                      | Yes.           |
+| `group.field`            | The field to be grouped.                                                                                      | Yes.           |
+| `group.limit`            | The maximum number of items allowed in the group.                                                             | Yes.           |
+| `group.ngroups`          | The number of groups that have matched the query in the results. The default value is false.                   | No.            |
+| `defType`                | Selects the query parser to be used to process the query.                                                    | No.            |
+| `q.op`                   | Specifies the default operator for query expressions, overriding the default operator specified in the Schema. Possible values are "AND" or "OR".                                                                                       | No.            |
+| `qf`                     | The query fields.                                                                                             | No.            |
```

Comment:
I think we should break this into two tables, one is the api parameters and the other is query parameters so its easy to understand the grouping of each.

Reply:
Changed